api/ttn/lorawan/v3/user.proto

```diff
@@ -553,6 +553,21 @@ message ListUserBookmarksRequest {
 
   // Only return recently deleted bookmarks.
   bool deleted = 5;
+
+  // Specifies that only bookmarks that reference the given entity type should be returned. If empty the filter is not
+  // applied.
+  string has_entity_type = 6 [(validate.rules).string = {
```

Comment:
Should we make this a `repeated` field called `entity_types` instead ? The backend already uses `in` so might as well expose that too.

Reply:
Well while I did wrote the store and registry logic with multiple values in mind, when converting the query parameters to a slice of string it ends up with all the strings inside the an array of 1 element `["item_x item_y]"` instead of `["item_x", "item_y"]`.

I tried to delve into how the fieldmask does its parsing but ended up opting for keeping the API with a single value for the field while leaving the implementation flexible for more values.

Reply:
The correct way to encode repeated fields as `GET` parameters is to repeat the pair like `key=value1&key=value2`.